### ISSUE TITLE
Opaque error for text/plain responses

### ISSUE THREAD

**Issue by SpComb** [kontena/pharos-kube-client#1]
If the HTTP API returns a `Content-Type: text/plain` response, then the error should show what request failed and with what response.

```
I, [2018-07-13T13:43:39.737062 #1]  INFO -- Pharos::Kube::Transport<https://167.99.39.233:6443>: GET /apis/extensions/v1beta1 => HTTP 200: <Pharos::Kube::API::MetaV1::APIResourceList> in 0.047s
I, [2018-07-13T13:43:39.787304 #1]  INFO -- Pharos::Kube::Transport<https://167.99.39.233:6443>: GET /apis/extensions/v1beta1/namespaces/kube-system/deployments/kube-dns => HTTP 200: <Pharos::Kube::Resource> in 0.049s
I, [2018-07-13T13:43:39.852477 #1]  INFO -- Pharos::Kube::Transport<https://167.99.39.233:6443>: PUT /apis/extensions/v1beta1/namespaces/kube-system/deployments/kube-dns <Pharos::Kube::Resource> => HTTP 200: <Pharos::Kube::Resource> in 0.063s
    Completed Configure DNS @ 167.99.39.233 in 0.164s
==> Configure Calico network @ 167.99.39.233
Invalid response Content-Type: text/plain; charset=utf-8
/usr/local/bundle/bundler/gems/pharos-kube-client-256e580aa10d/lib/pharos/kube/transport.rb:129:in `parse_response'
/usr/local/bundle/bundler/gems/pharos-kube-client-256e580aa10d/lib/pharos/kube/transport.rb:157:in `request'
/usr/local/bundle/bundler/gems/pharos-kube-client-256e580aa10d/lib/pharos/kube/transport.rb:195:in `get'
/usr/local/bundle/bundler/gems/pharos-kube-client-256e580aa10d/lib/pharos/kube/api_client.rb:41:in `api_resources'
/usr/local/bundle/bundler/gems/pharos-kube-client-256e580aa10d/lib/pharos/kube/api_client.rb:51:in `resource'
/app/lib/pharos/phases/configure_calico.rb:28:in `get_ippool'
/app/lib/pharos/phases/configure_calico.rb:35:in `validate_ippool'
/app/lib/pharos/phases/configure_calico.rb:45:in `validate'
/app/lib/pharos/phases/configure_calico.rb:49:in `call'
/app/lib/pharos/phase_manager.rb:70:in `block in apply'
/app/lib/pharos/phase_manager.rb:40:in `block in run_serial'
/app/lib/pharos/phase_manager.rb:39:in `map'
/app/lib/pharos/phase_manager.rb:39:in `run_serial'
/app/lib/pharos/phase_manager.rb:51:in `run'
/app/lib/pharos/phase_manager.rb:67:in `apply'
/app/lib/pharos/cluster_manager.rb:119:in `apply_phase'
/app/lib/pharos/cluster_manager.rb:99:in `apply_phases'
/app/lib/pharos/up_command.rb:104:in `configure'
/app/lib/pharos/up_command.rb:44:in `block in execute'
/app/lib/pharos/up_command.rb:43:in `chdir'
/app/lib/pharos/up_command.rb:43:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:132:in `run'
/app/lib/pharos/root_command.rb:16:in `run'
bin/pharos-cluster:12:in `<top (required)>'
```

**Comment by SpComb** [comment]
In this case, this is a 404 response for a non-existing CRD apigroup:

```
excon.response
  :body          => "404 page not found\n"
  :cookies       => [
  ]
  :headers       => {
    "Content-Length"         => "19"
    "Content-Type"           => "text/plain; charset=utf-8"
    "Date"                   => "Fri, 13 Jul 2018 13:50:59 GMT"
    "X-Content-Type-Options" => "nosniff"
  }
  :host          => "167.99.39.233"
  :local_address => "172.17.0.2"
  :local_port    => 45074
  :path          => "/apis/crd.projectcalico.org/v1"
  :port          => 6443
  :reason_phrase => "Not Found"
  :remote_ip     => "167.99.39.233"
  :status        => 404
  :status_line   => "HTTP/1.1 404 Not Found\r\n"
```

**Comment by SpComb** [comment]
Fixed in f0b6fe0
